addons/website_slides/models/slide_channel.py

```diff
@@ -533,26 +517,48 @@ def _compute_has_requested_access(self):
         for channel in self:
             channel.has_requested_access = channel.id in requested_cids
 
-    @api.depends('channel_partner_all_ids.partner_id', 'channel_partner_all_ids.member_status')
+    @api.depends('channel_partner_all_ids.partner_id', 'channel_partner_all_ids.member_status', 'channel_partner_all_ids.active')
     @api.depends_context('uid')
-    @api.model
     def _compute_membership_values(self):
         if self.env.user._is_public():
             self.is_member = False
             self.is_member_invited = False
             return
 
-        channel_partners = self.env['slide.channel.partner'].sudo().search([
-            ('channel_id', 'in', self.ids),
-            ('partner_id', '=', self.env.user.partner_id.id),
-        ])
-        active_channel_partners = channel_partners.filtered(
-            lambda channel_partner: channel_partner.member_status != 'invited')
-        invitation_pending_channels = (channel_partners - active_channel_partners).channel_id
-        active_channels = active_channel_partners.channel_id
+        data = {
+            member_status: channel_ids
+            for member_status, channel_ids in self.env['slide.channel.partner'].sudo()._read_group(
+                [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', 'in', self.ids), ('active', '=', True)],
+                ['member_status'], ['channel_id:array_agg']
```

Comment:
Continuing: seems to be a good idea to extract this part somehow

Reply:
I could do a bottleneck method but I would have to differentiate both domain, aggregate and processing. Unless I aggregate by member_status all the time and return both invited_channels and member_channel for all three methods, or differentiate return? Maybe not optimal for search methods?

So it could become something like
```python
    def _search_is_member_channel_ids(self, channel_ids, invited=False)
        domain = [('partner_id', '=', self.env.user.partner_id.id), ('active', '=', True)]
        if channel_ids:
            domain.append(('channel_id', 'in', channel_ids))
        data = {
            member_status: channel_ids
            for member_status,channel_ids in self.env['slide.channel.partner'].sudo()._read_group(
                domain,
                groupby=['member_status'],
                aggregates=['channel_id:array_agg']
        )
        member_channel_ids = data.get('joined', []) + data.get('ongoing', []) + data.get('completed', [])
        invitation_channel_ids = data.get('invited', [])
        return member_channel_ids, invitation_channel_ids if channel_ids else \
            invited_channel_ids if invited else member_channel_ids
```

What do you prefer? Atm, I've implemented a small method to extract the search from the two search methods.